cli/cmd/deploy/deploy.go

```diff
@@ -0,0 +1,397 @@
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/fatih/color"
+	"github.com/go-git/go-git/v5"
+	"github.com/joho/godotenv"
+	"github.com/rilldata/rill/admin/client"
+	"github.com/rilldata/rill/cli/cmd/auth"
+	"github.com/rilldata/rill/cli/cmd/cmdutil"
+	"github.com/rilldata/rill/cli/cmd/org"
+	"github.com/rilldata/rill/cli/cmd/project"
+	"github.com/rilldata/rill/cli/pkg/config"
+	"github.com/rilldata/rill/cli/pkg/dotrill"
+	"github.com/rilldata/rill/cli/pkg/gitutil"
+	"github.com/rilldata/rill/cli/pkg/variable"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type options struct {
+	Name        string
+	Description string
+	ProdBranch  string
+	Public      bool
+	Variables   []string
+
+	// projectPath string input taken interactively
+	// no input taken from user for below
+	region   string
+	dbDriver string
+	dbDSN    string
+	slots    int64
+}
+
+// DeployCmd is the guided tour for deploying rill projects to rill cloud.
+// TODO :: add non interactive mode
+func DeployCmd(cfg *config.Config) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "deploy",
+		Short: "Guided tour for deploying rill projects to rill cloud",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			warn := color.New(color.Bold).Add(color.FgYellow)
+			info := color.New(color.Bold).Add(color.FgWhite)
+			success := color.New(color.Bold).Add(color.FgGreen)
+			var adminClient *client.Client
+
+			// log in if not logged in
+			if !cfg.IsAuthenticated() {
+				msg := fmt.Sprintf("In order to deploy to Rill Cloud, you must login. Opening your browsers to %s to login or sign up...", cfg.AdminURL)
+				warn.Println(msg)
```

Comment:
1. Should it sleep for ~2 secs before auto-opening the browser so people have time to see the message?
2. Let's not include `cfg.AdminURL` in the message here – it's the backend URL, not the frontend URL. Also, the `LoginCmd` will print any URL the user needs to access.

Reply:
1. Ideally IMO we should always take input from user before auto opening browser.
I am okay with few seconds sleep as well so that people can read messages. But there the question will be how many seconds should be considered enough. Also if we add sleep here we need to add such sleep before all redirects.

Reply:
I'm also not a fan of opening the browser unexpectedly. But should we try with a sleep and see what people think? You can try out a few different delays.